include/circt/Dialect/SV/SVExpressions.td

```diff
@@ -204,7 +204,7 @@ def LocalParamOp : SVOp<"localparam",
   let results = (outs HWValueType:$result);
 
   let assemblyFormat = [{
-    `:` qualified(type($result)) `` custom<SVImplicitSSAName>(attr-dict)
+    `:` qualified(type($result)) custom<ImplicitSSAName>($name) attr-dict
```

Comment:
Nice, I didn't know that we can only pass $name :+1:

Reply:
I remember the first `` was intentionally introduced by @rwy7  to remove extra whitespaces. Do you have thoughts here? @rwy7

Reply:
Oh I just realized that we are putting the name and attr-dict after the type specifier, which seems a bit out-of-line with the rest of the code base.

I think the extra space was needed because the printer packs the `{` of attr-dict tightly to what comes before, but the `name "foo"` should get a proper space automatically. Not totally sure how the printer magic works though.

Reply:
The generated print function automatically puts whitespace before custom directives. So if the custom directive has no output, you can end up with an extra space. I think attr-dict will print it's own leading whitespace, too (but only if there's output). so in the case where:
- the name is not printed
- the attr-dict is printed

If I have it right, you'll have a double-space before the attr-dict.

Reply:
This is the convention set by the older helper, maybe you can improve on it:

- The optional directive should print its own leading space if it has any output.
- The asm format should elide the automatic leading space with ``

Reply:
Awesome, thanks a lot for clarifying! I'll revert that :+1: